### PR TITLE
Grid refactor

### LINKED ISSUE #7518: Implement new Grid with Article Page
**Is your feature request related to a problem? Please describe.**
**Parent Issue** https://github.com/bbc/simorgh/issues/3686

To break down the work for migrating over to our new grid implementation, we should focus on doing this on a page type basis with this issue focusing on `ArticlePage`.

**Describe the solution you'd like**
To be able to migrate over to the new grid, we need to replace all usage of [`GridItemConstrained`](https://github.com/bbc/simorgh/blob/latest/src/app/lib/styledGrid/index.jsx) in our containers to make use of the new grid. The containers used on `ArticlePage` are as follows:

- Headings (Shared with #7520 and #7519)
- ArticleMediaPlayer
- Text -> Paragraph
- Image (Shared with #7520 and #7519)
- ArticleTimestamp (Shared with #7520 and #7519)

We need to check if these containers are using the old grid and if so, we need migrate them over to the new implementation. Once the containers have been migrated we can then implement the new grid into `ArticlePage`. Some containers are shared with other page types, ensure that if those issues have been picked, that work isn't being duplicated.

A developer accessiblity swarm should be carried out on an article page when the new grid has been implemented before going live.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing. Regression checks are needed.
- Update snapshot tests

**Additional context**

### LINKED ISSUE #7519: Implement new Grid with Media Asset Page
**Is your feature request related to a problem? Please describe.**
**Parent Issue** https://github.com/bbc/simorgh/issues/3686

To break down the work for migrating over to our new grid implementation, we should focus on doing this on a page type basis with this issue focusing on `MediaAssetPage`.

**Describe the solution you'd like**
To be able to migrate over to the new grid, we need to replace all usage of [`GridItemConstrained`](https://github.com/bbc/simorgh/blob/latest/src/app/lib/styledGrid/index.jsx) in our containers to make use of the new grid. The containers used on `MediaAssetPage` are as follows:

- FauxHeadline (Shared with #7520)
- VisuallyHiddenHeadline (Shared with #7520)
- Headings (Shared with #7520 and #7518)
- CpsText (Shared with #7520)
- Image (Shared with #7520 and #7518)
- ArticleTimestamp (Shared with #7520 and #7518)
- CpsAssetMediaPlayer

We need to check if these containers are using the old grid and if so, we need migrate them over to the new implementation. Once the containers have been migrated we can then implement the new grid into `MediaAssetPage`. Some containers are shared with other page types, ensure that if those issues have been picked, that work isn't being duplicated. 

This issue and the issue for [PGL's](https://github.com/bbc/simorgh/issues/7520) have exactly the same containers used for blocks, so when work is carried out on either one, this should also complete the work for both.

A developer accessiblity swarm should be carried out on a MAP when the new grid has been implemented before going live.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing. Regression checks are needed.
- Update snapshot tests

**Additional context**

### LINKED ISSUE #7520: Implement new Grid with Photo Gallery Page
**Is your feature request related to a problem? Please describe.**
**Parent Issue** https://github.com/bbc/simorgh/issues/3686

To break down the work for migrating over to our new grid implementation, we should focus on doing this on a page type basis with this issue focusing on `PhotoGalleryPage`.

**Describe the solution you'd like**
To be able to migrate over to the new grid, we need to replace all usage of [`GridItemConstrained`](https://github.com/bbc/simorgh/blob/latest/src/app/lib/styledGrid/index.jsx) in our containers to make use of the new grid. The containers used on `PhotoGalleryPage` are as follows:

- FauxHeadline (Shared with #7520)
- VisuallyHiddenHeadline (Shared with #7520)
- Headings (Shared with #7520 and #7518)
- CpsText (Shared with #7520)
- Image (Shared with #7520 and #7518)
- ArticleTimestamp (Shared with #7520 and #7518)
- CpsAssetMediaPlayer - Not sure why PGL's have this, as I'm pretty sure they don't have videos

We need to check if these containers are using the old grid and if so, we need migrate them over to the new implementation. Once the containers have been migrated we can then implement the new grid into `PhotoGalleryPage`. Some containers are shared with other page types, ensure that if those issues have been picked, that work isn't being duplicated. 

This issue and the issue for [MAP's](https://github.com/bbc/simorgh/issues/7519) have exactly the same containers used for blocks, so when work is carried out on either one, this should also complete the work for both.

A developer accessiblity swarm should be carried out on a PGL when the new grid has been implemented before going live.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing. Regression checks are needed.
- Update snapshot tests

**Additional context**

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves #7518 
Resolves #7519 
Resolves #7520 

**Overall change:**
_A very high-level summary of easily-reproducible changes that can be understood by non-devs._

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by rebeccamcginn** [comment]
MAP LTR
http://localhost:7080/afrique/monde-49500638
https://www.test.bbc.com/afrique/monde-49500638?renderer_env=live

MAP RTL
http://localhost:7080/arabic/worldnews/2015/11/151120_t_arabic_av
https://www.test.bbc.com/arabic/worldnews/2015/11/151120_t_arabic_av

PGL LTR
http://localhost:7080/afaanoromoo/oduu-41217768
https://www.bbc.com/afaanoromoo/oduu-41217768

PGL RTL
http://localhost:7080/arabic/art-and-culture-38260491
https://www.test.bbc.com/arabic/art-and-culture-38260491?renderer_env=live

Article LTR
http://localhost:7080/afrique/articles/cz216x22106o
https://www.test.bbc.com/afrique/articles/cz216x22106o

Article RTL
http://localhost:7080/arabic/articles/c1er5mjnznzo
https://www.test.bbc.com/arabic/articles/c1er5mjnznzo

Most Watched LTR
http://localhost:7080/afrique/media/video
https://www.bbc.com/afrique/media/video

Most Watched RTL
http://localhost:7080/arabic/media/video
https://www.bbc.com/arabic/media/video

Most Read LTR
http://localhost:7080/afrique/popular/read
https://www.bbc.com/afrique/popular/read

Most Read RTL
http://localhost:7080/arabic/popular/read
https://www.bbc.com/arabic/popular/read

File: src/app/components/Grid/index.jsx [hunk 1]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
   </StyledCPSPageGrid>
 );
 
+export const ArticlePageGrid = ({ as, ...props }) => (
```
Comment (jroebu14) [inline src/app/components/Grid/index.jsx]: Looks like `ArticlePageGrid` is the same as `GridWrapper` so I'd remove `ArticlePageGrid` and just use `GridWrapper`

File: src/app/components/Grid/index.jsx [hunk 2]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+      group1: 6,
+      group2: 6,
+      group3: 6,
+      group4: 8,
+      group5: 20,
+    }}
+  />
+);
+
+export const GridItemConstrainedSmall = styled(({ as, dir, ...props }) => (
```
Comment (jroebu14) [inline src/app/components/Grid/index.jsx]: If we are renaming these I'd just take the word `Constrain` out

File: src/app/components/Grid/index.jsx [hunk 3]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+GridItemConstrainedLarge.propTypes = {
+  // eslint-disable-next-line react/require-default-props
+  as: string,
+  // eslint-disable-next-line react/require-default-props
+  dir: string,
+};
+
+GridItemConstrainedLargeNoMargin.propTypes = {
+  // eslint-disable-next-line react/require-default-props
+  as: string,
```
Comment (jroebu14) [inline src/app/components/Grid/index.jsx:297]: to remove the `// eslint-disable-next-line react/require-default-props` I think the default as prop could be `undefined`. The dir could be `ltr`

File: src/app/lib/layoutGrid/index.js [hunk 4]
```diff
@@ -1,9 +1,7 @@
 import { css } from 'styled-components';
```
Comment (jroebu14) [inline src/app/lib/layoutGrid/index.js]: would be good to eventually delete this file

File: src/app/lib/styledGrid/index.jsx [hunk 5]
```diff
@@ -1,24 +1,7 @@
-import { node, number } from 'prop-types';
```
Comment (jroebu14) [inline src/app/lib/styledGrid/index.jsx:1]: would be good to also delete this file

File: src/app/components/Grid/index.jsx [hunk 6]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+    {...props}
+    forwardedAs={as}
+    dir={dir}
+    item
+    startOffset={{
+      group0: 1,
+      group1: 1,
+      group2: 1,
+      group3: 1,
+      group4: 3,
```
Comment (RichardPK) [inline src/app/components/Grid/index.jsx]: At the 1008 - 1280 breakpoint, the `grid-column-start` value is set to 3, which un-centres everything:
Screenshot 2020-10-26 at 15 50 41. 

Updating the `grid-column-start` value to 2 for this breakpoint fixes this across the board.

```suggestion
      group4: 2,
```

File: src/app/components/Grid/index.jsx [hunk 7]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+      group3: true,
+      group4: false,
+      group5: false,
+    }}
+    startOffset={{
+      group0: 1,
+      group1: 1,
+      group2: 1,
+      group3: 1,
+      group4: 3,
```
Comment (RichardPK) [inline src/app/components/Grid/index.jsx]: See earlier comment RE the start column at this breakpoint

```suggestion
      group4: 2,
```

File: src/app/components/Grid/index.jsx [hunk 8]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+    {...props}
+    forwardedAs={as}
+    dir={dir}
+    item
+    startOffset={{
+      group0: 1,
+      group1: 1,
+      group2: 1,
+      group3: 1,
+      group4: 3,
```
Comment (RichardPK) [inline src/app/components/Grid/index.jsx]: See earlier comment RE the start column at this breakpoint

```suggestion
      group4: 2,
```

File: src/app/components/Grid/index.jsx [hunk 9]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+      group3: true,
+      group4: false,
+      group5: false,
+    }}
+    startOffset={{
+      group0: 1,
+      group1: 1,
+      group2: 1,
+      group3: 1,
+      group4: 3,
```
Comment (RichardPK) [inline src/app/components/Grid/index.jsx]: See earlier comment RE the start column at this breakpoint

```suggestion
      group4: 2,
```

File: src/app/components/Grid/index.jsx [hunk 10]
```diff
@@ -46,8 +102,275 @@ export const CPSPageGrid = ({ children, ...props }) => (
...
+    {...props}
+    forwardedAs={as}
+    dir={dir}
+    item
+    startOffset={{
+      group0: 1,
+      group1: 1,
+      group2: 1,
+      group3: 1,
+      group4: 3,
```
Comment (RichardPK) [inline src/app/components/Grid/index.jsx]: See earlier comment RE the start column at this breakpoint

```suggestion
      group4: 2,
```

File: src/app/containers/CpsOnwardJourney/index.jsx [hunk 11]
```diff
@@ -185,11 +185,11 @@ const CpsOnwardJourney = ({
         {children}
       </Wrapper>
     ) : (
-      <LegacyGridWrapper data-e2e={labelId} {...a11yAttributes}>
-        <LegacyGridItemConstrainedLarge>
+      <GridWrapper data-e2e={labelId} {...a11yAttributes}>
```
Comment (RichardPK) [inline src/app/containers/CpsOnwardJourney/index.jsx:188]: This element has a 2rem `padding-bottom` style on live that persists across all breakpoints:
Screenshot 2020-10-26 at 16 02 33

You can see it disappearing in ln6 of the snap: [src/app/containers/CpsOnwardJourney/__snapshots__/index.test.jsx.snap](https://github.com/bbc/simorgh/pull/8194/files#diff-8ffb0649a2a22986255f1fd00bad9f0be2eafcda4cf109f4e1d6aa0aee9dee9bL6)

I think you _could_ just make use of styled/emotion's `css` prop here to apply the style:

```suggestion
      <GridWrapper 
        data-e2e={labelId} 
        {...a11yAttributes} 
        css={css`
          padding-bottom: 2rem;
        `}
      >
```

**Review by RichardPK — COMMENTED** [review]
This is badass ppl thank you so much 👏 . I've included code suggestions to fix 2 visual bugs.

Only other visual difference is the max-width of the grid (with content on the screen filling a slightly larger area than before), though the new value is probably more correct than the current one.

Looks like the current content width is `78rem`, whereas `GelPageGrid` sets it to `80rem`: https://github.com/bbc/simorgh/blob/75df826662c6920b8bb96da79b126f0879f4264a/src/app/components/Grid/index.jsx#L74

No doubt 80rem is actually correct, just highlighting as it's likely this slight visual difference will get picked up!

Bravo again 🦖

File: src/app/containers/ArticleMediaPlayer/__snapshots__/index.test.jsx.snap [hunk 12]
```diff
@@ -27,40 +27,52 @@ exports[`MediaPlayer Calls the amp media player 1`] = `
...
-    max-width: 37.75rem;
+    margin-left: 40%;
   }
 }
 
 @media (min-width:80rem) {
   .c0 {
-    grid-column: 6 / span 10;
-    max-width: 38.5rem;
+    margin-left: 40%;
```
Comment (simonsinclair) [inline src/app/containers/ArticleMediaPlayer/__snapshots__/index.test.jsx.snap:62]: Do we have idea why we're only seeing fallback styles in these snapshots? I.E. No reference to `grid-column: 6 / span 10;` as before?

File: src/app/lib/layoutGrid/index.js (hunk 4 above)
Comment (simonsinclair) [reply src/app/lib/layoutGrid/index.js]: Do we still depend on this file, @jroebu14?

**Review by simonsinclair — COMMENTED** [review]
This is super work, @jroebu14 and @rebeccamcginn. The implementation looks sound, and I'd be happy to see this get merged. Hopefully chat to you in the morning about the collective comments.

**Comment by LilyL0u** [comment]
Problem with embeds on http://localhost:7080/mundo/23263889

Screenshot 2020-10-26 at 17 12 13

File: src/app/lib/styledGrid/index.jsx (hunk 5 above)
Comment (jroebu14) [reply src/app/lib/styledGrid/index.jsx:1]: Deleted!

File: src/app/lib/layoutGrid/index.js (hunk 4 above)
Comment (jroebu14) [reply src/app/lib/layoutGrid/index.js]: Deleted!

**Comment by LilyL0u** [comment]
Fixed bugs:
Resizing windows now keeps the media player centred
Padding underneath CPS onwards journey components now matches test
Includes display same as on test

**Review by RichardPK — APPROVED** [review]
LGTM :shipit:

**Comment by LilyL0u** [comment]
PGL - RTL text not right aligned

Image 27-10-2020 at 14 26

**Review by ryanmccombe — APPROVED** [review]
The `dir` fix should have affected some snapshots

Looks great though!

**Review by simonsinclair — APPROVED** [review]

**Review by ryanmccombe — APPROVED** [review]

**Comment by LilyL0u** [comment]
Here is a link to the spreadsheet documenting our testing of all relevant page types, LTR and RTL, canonical and amp over 28th-29th October: https://docs.google.com/spreadsheets/d/1XcvUahodb2wLI1Jaji6VsuQbZ8Ld4cDeUZi5iJZ-v1Q/edit#gid=0

**Pull request merged**